src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java

```diff
@@ -1,12 +1,510 @@
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ZendeskInputPlugin implements InputPlugin
 {
-    public ZendeskInputPlugin()
+    public interface PluginTask extends Task
+    {
+        @Config("login_url")
+        String getLoginUrl();
+
+        @Config("auth_method")
+        @ConfigDefault("basic")
+        AuthenticationMethod getAuthenticationMethod();
+
+        @Config("target")
+        Target getTarget();
+
+        @Config("username")
+        @ConfigDefault("null")
+        Optional<String> getUsername();
+
+        @Config("password")
+        @ConfigDefault("null")
+        Optional<String> getPassword();
+
+        @Config("token")
+        @ConfigDefault("null")
+        Optional<String> getToken();
+
+        @Config("access_token")
+        @ConfigDefault("null")
+        Optional<String> getAccessToken();
+
+        @Config("start_time")
+        @ConfigDefault("null")
+        Optional<String> getStartTime();
+
+        @Min(1)
+        @Max(30)
+        @Config("retry_limit")
+        @ConfigDefault("5")
+        int getRetryLimit();
+
+        @Min(1)
+        @Max(3600)
+        @Config("retry_initial_wait_sec")
+        @ConfigDefault("5")
+        int getRetryInitialWaitSec();
+
+        @Min(30)
+        @Max(3600)
+        @Config("max_retry_wait_sec")
+        @ConfigDefault("60")
+        int getMaxRetryWaitSec();
+
+        @Min(3)
+        @Max(100)
+        @Config("connection_timeout")
+        @ConfigDefault("30")
+        int getConnectionTimeout();
+
+        @Config("incremental")
+        @ConfigDefault("true")
+        boolean getIncremental();
+
+        @Config("includes")
+        @ConfigDefault("[]")
+        List<String> getIncludes();
+
+        @Config("dedup")
+        @ConfigDefault("true")
+        boolean getDedup();
+
+        @Config("app_marketplace_integration_name")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceIntegrationName();
+
+        @Config("app_marketplace_org_id")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceOrgId();
+
+        @Config("app_marketplace_app_id")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceAppId();
+
+        @Config("previous_records")
+        @ConfigDefault("[]")
+        List<String> getPreviousRecords();
+
+        @Config("columns")
+        SchemaConfig getColumns();
+
+        void setStartTime(Optional<String> startTime);
+
+        void setIncremental(boolean incremental);
```

Comment:
Remove this logic please. It's highly unrecommended to do so, ie. updating user's input. Besides, `incremental` mode is not equivalent to `whether there is Incremental API for the target`.